osfoffline/application/background.py

```diff
@@ -62,31 +62,30 @@ def get_current_user(self):
         return session.query(models.User).filter(models.User.logged_in).one()
 
     def stop_loop(self, close=False):
+        """ WARNING: Only pass in 'close' if you plan on creating an entirely new loop afterwards
+        """
         logging.info('stop loop')
         if self.loop.is_closed():
-            logging.info('loop already closed')
-
-        elif not self.loop.is_running():
-            logging.info('loop is stopped already. closing it')
-            self.loop.close()
-        else:
-            # stop loop when current tasks finish.
-            self.loop.call_soon(self.loop.stop)
-            logging.info('call_soon to loop.stop. will stop when polling/observing events finish.')
-            # todo: find better way?
+            logging.info('loop already stopped')
             if close:
-                while not self.loop.is_closed():
-                    if not self.loop.is_running():
-                        self.loop.close()
+                logging.info('closing loop')
```

Comment:
This particular log statement may be unnecessary (log that it's closing if loop is closed)